src/constructs/stack-set/stack-set.ts

```diff
@@ -142,7 +142,9 @@ export class GuStackSet extends CfnStackSet {
     const stackSetInstanceParameters = props.stackSetInstanceParameters ?? {};
 
     const params = Object.keys(stackSetInstanceParameters);
-    const undefinedStackSetParams = props.stackSetInstance.parameterKeys.filter((_) => !params.includes(_));
+    const parameterKeys = Object.keys(props.stackSetInstance.parameters);
+
+    const undefinedStackSetParams = parameterKeys.filter((_) => !params.includes(_));
 
     if (undefinedStackSetParams.length !== 0) {
       throw new Error(`There are undefined stack set parameters: ${undefinedStackSetParams.join(", ")}`);
```

Comment:
Our current model is to feed stack set instance's parameters from the parent stack, rather than have the stack set instance read from parameter store, as the orchestration of this is tricky.

Here we check that all parameters on the stack set instance also exist on the parent stack, throwing if not.